lib/posix/signal.c

```diff
@@ -77,7 +77,7 @@ int sigismember(const sigset_t *set, int signo)
 
 char *strsignal(int signum)
 {
-	static char buf[sizeof("RT signal " STRINGIFY(SIGRTMAX))];
+	static char buf[sizeof("RT signal " STRINGIFY(-INT_MAX))];
```

Comment:
Would `INT_MIN` be better than `-INT_MAX`?

Reply:
yeah, if the goal is silencing the warning because the compiler can't tell what the range of the possible values is, then the right answer is INT_MIN (one less than -INT_MAX, ofc)

Reply:
I initially tried to use `STRINGIFY(INT_MIN)`, but the compiler resolves that to `(-2147483647 - 1)` instead of `-2147483648`
so I resorted to the current workaround

```c
printk("-INT_MAX size: %d\n", sizeof("RT signal " STRINGIFY(-INT_MAX)));
printk("INT_MIN size: %d\n", sizeof("RT signal " STRINGIFY(INT_MIN)));
```

```log
-INT_MAX size: 22
INT_MIN size: 28
```

is there any better way to make this less confusing? maybe just
```c
static char buf[sizeof("RT signal -2147483648")];
```

?

Reply:
oh, right, INT_MIN isn't defined as a simple number. No, I'd go ahead and use your original `-INT_MAX` as that works fine for any size integers I can think of. A comment might be good though, just to make it clear why it's not using `INT_MIN`

Reply:
@keith-packard I have updated the patch to

```c
sizeof("RT signal -" STRINGIFY(INT_MAX))
```

instead and wrote the reasoning into the commit message, is that good enough?

Reply:
Comment in the code means someone looking at it won't get confused. Commit message is good, both even better.